invenio_rdm_records/services/communities/service.py

```diff
@@ -24,6 +25,38 @@ def create(self, identity, data, record, uow=None):
         pass
 
     @unit_of_work()
-    def delete(self, identity, id_, revision_id=None, uow=None):
+    def delete(self, identity, record_id, data, revision_id=None, uow=None):
         """Remove communities from the record."""
-        pass
+        record = self.record_cls.pid.resolve(record_id)
+        self.require_permission(identity, "remove_community_from_record", record=record)
+
+        valid_data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=True,
+        )
+
+        removed_communities = []
+        errored_communities = []
+        community_ids = valid_data["communities"]
+        for community_id in community_ids:
+            if community_id not in record.parent.communities.ids:
+                errored_communities.append(
+                    {
+                        "community_id": community_id,
+                        "error": f"The record {record_id} does not belong to the community {community_id}",
+                    }
+                )
+                continue
+            record.parent.communities.remove(community_id)
```

Comment:
is the `default` handled internally in the system field? What happens when the list of communities becomes `[]`? Worth adding a comment?

Reply:
Yes, it's already handled see https://github.com/inveniosoftware/invenio-communities/blob/master/invenio_communities/records/records/systemfields/communities/manager.py#L99